src/monolithic.md

```diff
@@ -0,0 +1,32 @@
+# Monolithic Architecture
+
+Blockchains as we know them have four functions. In no particular order:
+
+- Execution: Execute transactions to make updates to the state.
+- Settlement: Dispute resolution.
+- Consensus: Defines the state and validates that all nodes on the blockchain have the same state.
+- Data availability: Ensure data stored in each block is available.
+
+Monolithic blockchains are a type of blockchain architecture that handle all four functions, at the same time, on this single layer.
+
+![monolithic](./images/monolithic.png)
+
+## Challenges with Monolithic
+
+Some constraints and challenges with a monolithic architecture:
+
+### Costly and inefficient transaction verification
+
+In order to verify the validity of transactions in the chain, full nodes must download the entire chain and execute each transaction locally.
+
+### Not designed for fraud proofs
+
+Using the general construction for fraud proofs, in order to create a fraud proof you have to provide the pre-state and compute the post-state, comparing what the block producer's post-state was to their outcome. If a contract calls other contracts, the pre-state could be extremely large, and the cost to produce a fraud proof could be unbound.
```

Comment:
This section, while true in isolation, isn't really a related to monolithic vs modular. I would instead have a section after the next section (or expand the next section) on how using up the node's capacity for execution means there's less capacity left over for data availability.